CI build failed on protocol reading due to lock

## Description

[Merge build failed](https://github.com/camunda/camunda/actions/runs/9226120982/job/25385192985) with obscure error:

```
  Failed to execute goal com.salesforce.servicelibs:proto-backwards-compatibility:1.0.7:backwards-compatibility-check (default) on project zeebe-gateway-protocol: An error occurred while running protolock: Cannot run program "/home/runner/work/camunda/camunda/zeebe/gateway-protocol/target/protolock-bin/protolock" (in directory "/home/runner/work/camunda/camunda/zeebe/gateway-protocol"): error=26, Text file busy -> [Help 1]
Error:  
```

Looks like there were multiple builds referencing the same file?

Happened again: https://github.com/camunda/camunda/actions/runs/9842149374/job/27170461841
```
 Failed to execute goal com.salesforce.servicelibs:proto-backwards-compatibility:1.0.7:backwards-compatibility-check (default) on project zeebe-gateway-protocol: An error occurred while running protolock: Cannot run program "/home/runner/work/camunda/camunda/zeebe/gateway-protocol/target/protolock-bin/protolock" (in directory "/home/runner/work/camunda/camunda/zeebe/gateway-protocol"): error=26, Text file busy -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.salesforce.servicelibs:proto-backwards-compatibility:1.0.7:backwards-compatibility-check (default) on project zeebe-gateway-protocol: An error occurred while running protolock
<..>
Caused by: java.io.IOException: Cannot run program "/home/runner/work/camunda/camunda/zeebe/gateway-protocol/target/protolock-bin/protolock" (in directory "/home/runner/work/camunda/camunda/zeebe/gateway-protocol"): error=26, Text file busy
```

https://github.com/camunda/camunda/actions/runs/10561422487/job/29257143645

```
Caused by: java.io.IOException: Cannot run program "/runner/_work/camunda/camunda/zeebe/gateway-protocol/target/protolock-bin/protolock" (in directory "/runner/_work/camunda/camunda/zeebe/gateway-protocol"): error=26, Text file busy 
```

--- a/buf.yaml
+++ b/buf.yaml
@@ -1,0 +1,14 @@
+version: v2
+modules:
+  - path: zeebe/dynamic-config/src/main/resources/proto
+    breaking:
+      # as we use this only internally, only break on wire incompatibility; ignore source breaking
+      # changes
+      use:
+        - WIRE
+  - path: zeebe/gateway-protocol/src/main/proto
+    breaking:
+      # as this is meant to be consumed by clients for code generation, we do care about source
+      # backwards compatibility
+      use:
+        - FILE

